agent/engine/docker_task_engine.go

```diff
@@ -446,12 +447,24 @@ func (engine *DockerTaskEngine) AddTask(task *api.Task) error {
 
 	existingTask, exists := engine.state.TaskByArn(task.Arn)
 	if !exists {
+		// This will update the container desired status
+		task.UpdateDesiredStatus()
+
 		engine.state.AddTask(task)
-		engine.startTask(task)
-	} else {
-		engine.updateTask(existingTask, task)
+		if dependencygraph.ValidDependencies(task) {
+			engine.startTask(task)
+		} else {
+			seelog.Errorf("Task cannot move forward due to the dependencies of containers cannot be resolved, task: %s", task.String())
+			task.SetKnownStatus(api.TaskStopped)
+			task.SetDesiredStatus(api.TaskStopped)
+			engine.emitTaskEvent(task, "impossibleTransitionError: Cannot resolve the dependencies of containers in the task definition")
```

Comment:
Apologies, I should have made this clearer in my earlier comment. Please use the `impossibleTransitionError ` type instead: https://github.com/aws/amazon-ecs-agent/blob/f88e52ec2d20fd65b2ccffc0019cc1aa203e487d/agent/engine/errors.go#L35 instead.